app/models/conversion_host.rb

```diff
@@ -12,14 +10,16 @@ class ConversionHost < ApplicationRecord
 
   validates :name, :presence => true
   validates :resource_id, :presence => true
-  validates :resource_type, :presence => true, :inclusion => { :in => VALID_RESOURCE_TYPES }
+  validates :resource_type, :presence => true
```

Comment:
Pretty sure you can just `validated :resource, :presence => true` without checking the _id and _type individually:
```
ActiveRecord::RecordInvalid (Validation failed: ConversionHost: Resource can't be blank)
```

Reply:
Ok, updated.